src/java.base/linux/classes/jdk/internal/platform/CgroupInfo.java

```diff
@@ -107,7 +107,7 @@ public void setCgroupPath(String cgroupPath) {
      *
      */
     static CgroupInfo fromCgroupsLine(String line) {
-        String[] tokens = line.split("\\s+");
+        String[] tokens = line.split("\t");
```

Comment:
As far as I can tell, the delimiter hasn't changed since the file was introduced, and judging by the kernel mailing list (e.g., see [the following](https://lore.kernel.org/all/Yr5JVHhSUCrbT8OH@mtj.duckdns.org/)), I don't think it will change any time soon.

Reply:
I'm not convinced this is a good change. Going by that mailing list thread, it suggests that people considered changing it. If one of those attempts were successful, it would have broken this code. It makes it rather fragile. The issue, with container detection code going wrong is that you most likely never notice. Translating this to GraalVM means that the native image, would a) detect the wrong version in use or b) fail detection and use host values. In both cases the application will likely misbehave in a container setup with resource limits applied and you won't (easily) know why. So even though it's unlikely to be a problem, there is a chance it could be and it's asking for trouble for no good reason.

Therefore, being conservative about delimiters makes sense here. My choice in this case would be more robust code over relying on external factors. YMMV.

Reply:
Okay, so how about something like the following, would that be more acceptable?
```suggestion
        String[] tokens = Collections.list(new StringTokenizer(line, " \t")).toArray(new String[0]);
```